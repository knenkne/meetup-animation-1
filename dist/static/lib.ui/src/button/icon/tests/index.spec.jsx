import React from 'react'
import { mount, shallow } from 'enzyme'

import { IconButton } from '..'

describe('<Button.Icon />', () => {
    it('is available', () => {
        const wrapper = shallow(<IconButton title="title" icon="icon:core/operations/betweenAccounts" />)

        expect(wrapper).toMatchSnapshot()
    })

    it('title has classNames "sm", "green"', () => {
        const wrapper = shallow(<IconButton
            title="title"
            icon="icon:core/operations/betweenAccounts"
            size="sm"
            colorScheme="green"
        />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('span').at(0).hasClass(IconButton.theme.sm)).toBeTruthy()
        expect(wrapper.find('span').at(0).hasClass(IconButton.theme.green)).toBeTruthy()
    })

    it('is disabled when mode=loading', () => {
        const wrapper = shallow(<IconButton
            title="title"
            icon="icon:core/operations/betweenAccounts"
            mode="loading"
        />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('button').at(0).props().disabled).toBeTruthy()
    })

    it('fire onClick handler', () => {
        const clickHandler = jest.fn()
        const defaultProps = {
            onClick: clickHandler,
            title: 'title',
            icon: 'icon:core/operations/betweenAccounts'
        }
        const wrapper = mount(<IconButton {...defaultProps} />)
        expect(clickHandler).not.toHaveBeenCalled()
        wrapper.find('button').simulate('click')
        expect(clickHandler).toHaveBeenCalled()
    })

    it('does not fire onClick handler if disabled', () => {
        const clickHandler = jest.fn()
        const defaultProps = {
            onClick: clickHandler,
            title: 'title',
            icon: 'icon:core/operations/betweenAccounts',
            disabled: true
        }
        const wrapper = mount(<IconButton {...defaultProps} />)
        expect(clickHandler).not.toHaveBeenCalled()
        wrapper.find('button').simulate('click')
        expect(clickHandler).not.toHaveBeenCalled()
    })

    it('does not fire onClick handler if loading', () => {
        const clickHandler = jest.fn()
        const defaultProps = {
            onClick: clickHandler,
            title: 'title',
            icon: 'icon:core/operations/betweenAccounts',
            mode: 'loading'
        }
        const wrapper = mount(<IconButton {...defaultProps} />)
        expect(clickHandler).not.toHaveBeenCalled()
        wrapper.find('button').simulate('click')
        expect(clickHandler).not.toHaveBeenCalled()
    })
})
