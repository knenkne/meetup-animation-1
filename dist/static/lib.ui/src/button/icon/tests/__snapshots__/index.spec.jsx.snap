// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button.Icon /> is available 1`] = `
<button
  aria-busy={false}
  className="button"
  disabled={false}
  type="button"
>
  <Icon
    name="icon:core/operations/betweenAccounts"
    onClick={[Function]}
    onError={[Function]}
    size="self"
    theme={
      Object {
        "self": "self icon",
      }
    }
  />
  <span
    className="title black md"
  >
    title
  </span>
</button>
`;

exports[`<Button.Icon /> is disabled when mode=loading 1`] = `
<button
  aria-busy={true}
  className="button"
  disabled={true}
  mode="loading"
  type="button"
>
  <Loader.Icon
    colorScheme="dark"
    size="lg"
  />
  <span
    className="title black md"
  >
    title
  </span>
</button>
`;

exports[`<Button.Icon /> title has classNames "sm", "green" 1`] = `
<button
  aria-busy={false}
  className="button"
  disabled={false}
  type="button"
>
  <Icon
    name="icon:core/operations/betweenAccounts"
    onClick={[Function]}
    onError={[Function]}
    size="self"
    theme={
      Object {
        "self": "self icon",
      }
    }
  />
  <span
    className="title green sm"
  >
    title
  </span>
</button>
`;
